akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+If the serializer you are using is "structural", like for example JSON or Avro are, then renames must be performed
+explicitly and you may need to perform the mapping from old-named fields to the new ones when recovering old events.
+
+Schema based formats, like protobuf or thrift, however allow such rename operations to be performed "for free",
+because the field name is never actually stored in the binary representation of the message. This is one of the
+advantages of schema based serializers, even though that they add the overhead of having to maintain the schema.
+When using serializers like this, no additional code change (except renaming the field and method used during serialization)
+is needed to perform such evolution:
+
+.. figure:: ../images/persistence-serializer-rename.png
+   :align: center
+
+It is important to learn about the strengths and limitations of your serializers, in order to be able to move
+swiftly and refactor your models fearlessly as you go on with the project.
+
+.. note::
+  Learn in-depth about the serialization engine you're using as it will impact how you can aproach schema evolution.
+
+  Some operations are "free" in certain serialization formats (more often than not: removing/adding optional fields,
+  sometimes renaming fields etc.), while some other operations are strictly not possible.
+
+.. _remove-event-class-scala:
+
+Remove event class and ignore events
+------------------------------------
+
+**Situation:**
+While investigating app performance you notice that insane amounts of ``CustomerBlinked`` events are being stored
+for every customer each time he/she blinks. Upon investigation you decide that the event does not add any value
+and should be deleted. You still have to be able recover from a journal which contains those old CustomerBlinked events though.
+
+**Naive solution - drop events in EventAdapter:**
+
+The problem of removing an event type from the domain model is not as much it's removal, as the implications
+for the recovery mechanisms that this entails. For example, a naive way of filtering out certain kinds of events from
+being delivered to a recovering ``PersistentActor`` is pretty simple, as one can simply filter them out in an ``EventAdapter``:
+
+
+.. figure:: ../images/persistence-drop-event.png
+   :align: center
+
+   The ``EventAdapter`` can drop old events (**O**) by emitting an empty :class:`EventSeq`.
+   Other events can simply be passed through (**E**).
+
+This however does not address the underlying cost of having to deserialize all the events during recovery,
+even those which will be filtered out by the adapter. In the next section we will improve the above explained mechanism
+to avoid deserializing events which would be filtered out by the adapter anyway, thus allowing to save precious time
+during a recovery containing lots of such events (without actually having to delete them).
+
```

Comment:
This section is a bit bare since it  doesn't actually explain what an event adapter is or link to such an explanation. An example would be nice but if the concept of an event adapter is properly explained (or linked to) then the diagram is good enough I think.